File: src/handlers/review-handler.ts
import type { ReviewResult, StructuredError } from "../core/domain.js";
import type { DataStore } from "../core/ports.js";
import { review } from "../core/reviewer.js";

export interface ReviewHandlerInput {
  routeId?: string;
  project?: string;
  store: DataStore;
  meshReachable?: boolean;
}

export function handleReview(input: ReviewHandlerInput): ReviewResult | StructuredError {
  try {
    const result = review({
      routeId: input.routeId,
      project: input.project,
      identityId: input.store.getIdentityId(),
      store: input.store,
      meshReachable: input.meshReachable,
    });
    return result;
  } catch (e) {
    const message = e instanceof Error ? e.message : "Unknown error";
    return {
      error: { code: "INTERNAL", message: `Review failed: ${message}` },
    };
  }
}

File: src/openplan/handlers/review_handler.py
from openplan.core.reviewer import review_route


async def handle_review(conn, args: dict) -> dict:
    route_id = args.get("route_id")
    project = args.get("project")
    api_key = args.get("api_key", "")

    result = review_route(conn, route_id, project, api_key=api_key)
    return result
